google/cloud/forseti/notifier/notifiers/email_violations.py

```diff
@@ -76,8 +80,14 @@ def _write_temp_attachment(self):
         # Make attachment
         output_file_name = self._get_output_filename()
         output_file_path = '{}/{}'.format(TEMP_DIR, output_file_name)
-        with open(output_file_path, 'w+') as f:
-            f.write(parser.json_stringify(self.violations))
+        with csv_writer.write_csv(resource_name='violations',
+                                  data=self.violations,
+                                  write_header=True) as csv_file:
+            output_csv_name = csv_file.name
+            LOGGER.info('CSV filename: %s', output_csv_name)
+
+            shutil.copyfile(output_csv_name, output_file_path)
```

Comment:
Just to double-check, is the csv file being cleaned up after everything is done?

Reply:
Yes, it uses a `NamedTemporaryFile` that is deleted upon context manager exit.